src/video/vga_state.c

```diff
@@ -60,7 +60,7 @@ void vga_state_render(void) {
         for(unsigned int i = 0; i < state.transformer_count; i++) {
             state.transformers[i].callback(&tmp, &state.current, state.transformers[i].userdata);
         }
-        damage_copy(&state.dmg_current, &tmp);
+        damage_combine(&state.dmg_current, &tmp);
```

Comment:
Doesn't this cause the dmg_current to always include full palette, because the ranges can only grow ?

Reply:
no, because of `vga_state_mark_palette_flushed` (called from opengl3):
```c
void vga_state_mark_palette_flushed(void) {
    damage_reset(&state.dmg_current);
}
```